rust/lance-io/src/lib.rs

```diff
@@ -78,3 +78,27 @@ impl From<&Self> for ReadBatchParams {
         params.clone()
     }
 }
+
+impl ReadBatchParams {
+    pub fn to_offsets(&self, base_offset: u32, length: u32, total_num_rows: u32) -> Vec<u32> {
```

Comment:
It's a little unclear what these parameters are. Would you mind documenting them, and maybe adding a unit test below?

Reply:
This was a good exercise.  Turns out I was able to simplify the method a bit (get rid of `total_num_rows`).  I documented and added a test.